### DISCUSSION TITLE
WSJT-X 2 Cloudlog

### DISCUSSION THREAD

**Discussion by PA4RB** [magicbug/Cloudlog#899] [Show and tell]
Hi all,

I missed the interfacing with Cloudlog whilst working with WSJT-X. That's the reason why I built my own Windows application for that.

Some highlights:
- One time configuration. If you succeeded in configuring WSJT-X and/or Cloudlog this will be easy....
- Checks if WSJT-X is running
- Configurable timer for periods between reading the WSJT-X .adi file
- Uploads your latest QSO from WSJT-X to Cloudlog via API
- Monitor your latest QSO
- Only for Windows (Currently being developed on Win 10)
- No open source

![screenshot](https://user-images.githubusercontent.com/60810525/109422514-26768680-79dc-11eb-9872-f14ef8bc1c97.jpg)

I am actually in need for a few testers. If anyone is interested. Please let me know.

Rob, PD0RZH

**Comment by M0LTE** [comment]
Hi Rob, I’d recommend watching the WSJT-X UDP API instead of monitoring that log file. 

Is this .NET? If so, see this library:

https://github.com/M0LTE/WsjtxUdpLib
and
https://www.nuget.org/packages/WsjtxUdpLib/

My .NET Core console app adifpush does the same as your app. It’s open source and cross platform.

**Comment by soakland5** [comment]
Rob, Can you tell me where I can get you Windows app for this?

**Reply by PA4RB** [reply]
Hi you could try if it fits your needs. Program is called Log2 you can find it on http://pd0rzh.nl

**Reply by soakland5** [reply]
I downloaded it but it complains that it needs MSWINSCK. I am running Windows 11. Any way around this? I installed Visual basic but till no-go.

**Reply by soakland5** [reply]
Never mind....I tried running it as administrator and it worked. Thanks.

**Reply by PA4RB** [reply]
> Never mind....I tried running it as administrator and it worked. Thanks.

And? Did it work for you. If not, what was the issue?

Regards Rob

**Comment by soakland5** [comment]
Thanks! I’ll download it and try it out. Just what I’ve been looking for.
I’m kinda tired of having to run Gridtracker just to upload to Cloudlog

**Comment by PA4RB** [comment]
Your welcome

Op ma 27 feb. 2023 om 15:28 schreef soakland5 ***@***.***>

**Comment by soakland5** [comment]
Oh sorry yes it works beautifully. Thank you so
Much for writing this!

Scott W1AL

**Comment by phl0** [comment]
@soakland5 I did an hour of debugging the Windows stuff today: Besides the WSJT-X improved PLUS version you need SSL libraries. It is important to install the v1.1.x branch ant not the latest 3.x. The libs can be found on https://slproweb.com/products/Win32OpenSSL.html.

![FrVUHZiWcAAfyi3](https://user-images.githubusercontent.com/7112907/225585054-dd9d0358-c423-4934-a276-73a3ff1f881e.png)

I just tested this on Win 10 and 11 x64 platform and it works.

**Comment by soakland5** [comment]
Awesome, thanks! I’ll try it

**Reply by phl0** [reply]
> Awesome, thanks! I’ll try it

Looking forward to your feedback ... :)

**Comment by soakland5** [comment]
I am pleased to report that after following your instructions, wsjt-x
improved plus will now log to cloudlog perfectly. The green button even
comes on! Hi hi. Thanks so much !!

**Reply by phl0** [reply]
Thank you for the feedback. So it was worth investing the hour of debugging here. Sorry for the hassles with Windows but that is out of my reach. Sri.

**Comment by 5B4AHO** [comment]
Hi There....I amtrying to get your app to work for me, but cant get it to work.

I am running cloudlog on thier server, cant find station number.

Got your app installed but the Log2 box when running is not showingthe options shown in your post at the bottomof the box.....no options like wsjt-x is running etc.

It says " using cloudlog, initialisation successful." but when I make qso in WSJT-X and submit to log I Get

"Cloudlog: <status>200</status>
<response>{"status":"failed","reason":"wrong JSON"}</response>
<statustext>OK</statustext>


Using Cloudlog
Application initialisation succesful."

Thanks for any help

Baz
5B4AHO

**Comment by magicbug** [comment]
Not sure if you have tried it yet but there's also Cloudlog Aurora https://aurora.cloudlog.org/

To find the number click callsign top right in the menu to open the menu go to Station Locations then you'll see ID listed

**Reply by 5B4AHO** [reply]
Thanks for that!
Just installed it thanks....Working fine with WSJT-x
Think I will stick with that then!
Cheers
Baz
